src/bun.js/node/node_util_binding.zig

```diff
@@ -105,3 +105,70 @@ pub fn internalErrorName(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFr
     var fmtstring = bun.String.createFormat("Unknown system error {d}", .{err_int}) catch bun.outOfMemory();
     return fmtstring.transferToJS(globalThis);
 }
+
+/// `extractedSplitNewLines` for ASCII/Latin1 strings. Panics if passed a non-string.
+//Returns `undefined` if param is utf8 or utf16 and not fully ascii.
+///
+/// ```js
+/// // util.js
+/// const extractedNewLineRe = new RegExp("(?<=\\n)");
+/// extractedSplitNewLines = value => RegExpPrototypeSymbolSplit(extractedNewLineRe, value);
+/// ```
+pub fn extractedSplitNewLinesFastPathStringsOnly(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!JSC.JSValue {
+    var fallback = std.heap.stackFallback(1024, bun.default_allocator);
+    const allocator = fallback.get();
+    bun.assert(callframe.argumentsCount() == 1);
+    const value = callframe.argument(0);
+    bun.assert(value.isString());
+
+    const str = try value.toBunString2(globalThis);
+
+    // if (str.isUTF16()) return JSC.JSValue.jsUndefined();
+    if (str.is8Bit() or bun.strings.isAllASCII(str.byteSlice())) {
```

Comment:
`bun.strings.isAllASCII(str.byteSlice())` is reinterpreting bytes of utf16 data as utf8/ascii. for example, this check is wrong for the character `㈴`, which in UTF-16 would be `0x34 0x32`, interpreted as ascii gives you `42`

(imo, byteSlice should be removed to prevent this category of mistake)